modules/light-clients/07-tendermint/light_client_module_test.go

```diff
@@ -1020,3 +749,458 @@ func (suite *TendermintTestSuite) TestVerifyNonMembership() {
+		clientState *ibctm.ClientState
+	)
+
+	testCases := []struct {
+		name      string
+		malleate  func()
+		expStatus exported.Status
+	}{
+		{
+			"client is active",
+			func() {},
+			exported.Active,
+		},
+		{
+			"client is frozen",
+			func() {
+				clientState.FrozenHeight = clienttypes.NewHeight(0, 1)
+				path.EndpointA.SetClientState(clientState)
+			},
+			exported.Frozen,
+		},
+		{
+			"client status without consensus state",
+			func() {
+				newLatestHeight, ok := clientState.LatestHeight.Increment().(clienttypes.Height)
+				suite.Require().True(ok)
+				clientState.LatestHeight = newLatestHeight
+				path.EndpointA.SetClientState(clientState)
+			},
+			exported.Expired,
+		},
+		{
+			"client status is expired",
+			func() {
+				suite.coordinator.IncrementTimeBy(clientState.TrustingPeriod)
+			},
+			exported.Expired,
+		},
+		{
+			"client state not found",
+			func() {
+				store := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
+				store.Delete(host.ClientStateKey())
+			},
+			exported.Unknown,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.SetupClients()
+
+			lightClientModule, found := suite.chainA.App.GetIBCKeeper().ClientKeeper.Route(path.EndpointA.ClientID)
+			suite.Require().True(found)
+
+			var ok bool
+			clientState, ok = path.EndpointA.GetClientState().(*ibctm.ClientState)
+			suite.Require().True(ok)
+
+			tc.malleate()
+
+			status := lightClientModule.Status(suite.chainA.GetContext(), path.EndpointA.ClientID)
+			suite.Require().Equal(tc.expStatus, status)
+		})
+
+	}
+}
+
+func (suite *TendermintTestSuite) TestLatestHeight() {
+	var (
+		path   *ibctesting.Path
+		height exported.Height
+	)
+
+	testCases := []struct {
+		name      string
+		malleate  func()
+		expHeight exported.Height
+	}{
+		{
+			"success",
+			func() {},
+			clienttypes.Height{RevisionNumber: 0x1, RevisionHeight: 0x4},
+		},
+		{
+			"client state not found",
+			func() {
+				store := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
+				store.Delete(host.ClientStateKey())
+			},
+			clienttypes.ZeroHeight(),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.SetupClients()
+
+			lightClientModule, found := suite.chainA.App.GetIBCKeeper().ClientKeeper.Route(path.EndpointA.ClientID)
+			suite.Require().True(found)
+
+			tc.malleate()
+
+			height = lightClientModule.LatestHeight(suite.chainA.GetContext(), path.EndpointA.ClientID)
+			suite.Require().Equal(tc.expHeight, height)
+		})
+	}
+}
+
+func (suite *TendermintTestSuite) TestGetTimestampAtHeight() {
+	var (
+		path   *ibctesting.Path
+		height exported.Height
+	)
+	expectedTimestamp := time.Unix(1, 0)
```

Comment:
not sure if this variable is useful here?

Reply:
I think we could be using `suite.chainB.LatestCommittedHeader.Header.Time` instead of overwriting with a predefined timestamp below